File: ProblemSolve/532.cs
/****************************
 * K-diff Pairs in an Array *
 ****************************/

public class Solution {
    public int FindPairs(int[] nums, int k) {
        Dictionary<int, int> hash = new Dictionary<int, int>();
        HashSet<int> ansSet = new HashSet<int>();
        
        foreach(int num in nums){
            if(hash.ContainsKey(num - k)){
                ansSet.Add(num);
            }
            
            if(hash.ContainsKey(num + k)){
                ansSet.Add(num + k);
            }
            
            hash[num] = 1;
        }
        
        return ansSet.Count;
    }
}

File: ProblemSolve/532.cpp
/****************************
 * K-diff Pairs in an Array *
 ****************************/

class Solution {
public:
    int findPairs(vector<int>& nums, int k) {
        unordered_map<int, int> hash;
        unordered_set<int> ansSet;
        
        for(int i=0; i<nums.size(); ++i){
            if(hash.count(nums[i] - k)){
                ansSet.insert(nums[i]);
            }
            
            if(hash.count(nums[i] + k)){
                ansSet.insert(nums[i] + k);
            }

            hash[nums[i]]++; 
        }
        
        return ansSet.size();
    }
};